### PR TITLE
Use solidity fuzzy import parser

### PR DESCRIPTION
**Pull request by spalladino**
Increases performance of nothing-to-compile runs by using a custom fuzzy parser that only extracts imports from the text, instead of the full antlr4 parser.

Running both versions in the contracts repository, this yields a 40% speed increase when having nothing to compile.

This commit also lazy-loads truffle-flattener, which was causing significant delays to the CLI startup.

### CODE REVIEW

**Comment by spalladino** [comment]
We should wait until https://github.com/frangio/fuzzy-import-parser/issues/2 is implemented before merging

File: packages/cli/src/utils/solidity.ts [hunk 1]
```diff
@@ -35,15 +36,9 @@ export function compilerSettingsMatch(s1: CompilerVersionOptions, s2: CompilerVe
...
-  const importedFiles: string[] = [];
-  parser.visit(ast, {
-    ImportDirective: (node: { path: string }) => importedFiles.push(node.path),
-  });
-
-  return importedFiles;
+  try {
+    return getMetadata(source).imports;
+  } catch {
+    return [];
```
Comment (spalladino) [inline packages/cli/src/utils/solidity.ts]: I'd output a warning at least, just in case that the error is different

Comment (nventuro) [reply packages/cli/src/utils/solidity.ts]: Wouldn't that be a bug in the parser? It shouldn't crash when fed valid Solidity source

**Comment by spalladino** [comment]
Indeed. That's why I want to be able to report it and not silently fail.

**Comment by nventuro** [comment]
@spalladino Once CI passes, this should be ready to merge. I extracted the error handling out of `getImports` to make it more explicit, and added calls to `trimURL` whenever we report a problematic file location that may have been the result of buggy parsing.

**Comment by spalladino** [comment]
@nventuro given I'm the author, I cannot approve the PR. Please approve it and merge, I think it's ok!

**Pull request merged**